### DISCUSSION TITLE
AZ CLI Authentication

### DISCUSSION THREAD

**Discussion by robsissons-contino** [Azure/azqr#168] [Q&A] [answered]
The documentation refers to Azure CLI as a supported method of authentication, but if I log in with 'az login' (with a user object) then try to perform a scan using 'azqr scan -s {subId}' I receive an error advising that the tenant ID / client ID is not set.

Does the authentication to AZ CLI have to be with a service principal?

**Comment by cmendible** [comment]
Hi are you using the env variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET?

**Reply by robsissons-contino** [reply]
Hi, I was trying to first authenticate with "az login" and use the context of that authentication with the tool, is this not supported?

If I understand correctly then, I need to register an SPN and set the env variables above?

**Reply by cmendible** [reply, marked as answer]
if you want to use the CLI then use: `az login --service-principal -u <app-id> -p <password-or-cert> --tenant <tenant>` which should work without issues, unless you have one of the above env variables set. In such a case clear or remove the vars before attempting the scan...
